internal/policy/policy.go

```diff
@@ -39,10 +43,17 @@ func (p *Policy) validate() (err error) {
 	if err != nil {
 		return err
 	}
+
 	p.Destination, err = urlParse(p.To)
 	if err != nil {
 		return err
 	}
+
+	// Only allow public access if no other whitelists are in place
+	if p.AllowPublicUnauthenticatedAccess && (p.AllowedDomains != nil || p.AllowedGroups != nil || p.AllowedEmails != nil) {
```

Comment:
Could we add a test or two for this?

Reply:
I added two tests to `proxy/handlers_test.go` that should test for this. Code coverage report shows this branch is tested. Is there a more in-depth test you'd like to see?

Reply:
Maybe I've got this wrong but this but it shows as uncovered for me (this would be for policy, not handlers right?)

https://codecov.io/gh/pomerium/pomerium/compare/c53c07c274f23d97f24e56ab1e75e8a5a288b92c...0952a1de46ba5faf1828567c228d536c8a810775/src/internal/policy/policy.go#L54

Reply:
Sorry i meant `proxy/proxy_test.go`

Reply:
Ah got it, yeah it's weird that codecov isn't showing this as covered as 

```go
{"policy public and whitelist bad", publicWithWhitelistPolicy, true},
``` 

should be doing this right?

Reply:
Yeah, that line should be checking that. That's really weird. I see that branch covered after running the `TestOptions_Validate` function locally.

![image](https://user-images.githubusercontent.com/11142171/58217587-6837b080-7cd1-11e9-8a3c-c7b0cb3686c5.png)

Reply:
LGTM then 👍